services/rfq/relayer/relconfig/config.go

```diff
@@ -228,5 +233,34 @@ func (c Config) Validate() (err error) {
 			return fmt.Errorf("total initial percent does not total 100 for %s: %f", token, sum)
 		}
 	}
+
+	return nil
+}
+
+// ValidateTokenDecimals calls decimals() on the ERC20s to ensure that the decimals in the config match the actual token decimals.
+func (c Config) ValidateTokenDecimals(ctx context.Context, omniClient omniClient.RPCClient) (err error) {
+	for chainID, chainCfg := range c.Chains {
+		for tokenName, tokenCFG := range chainCfg.Tokens {
+			chainClient, err := omniClient.GetChainClient(ctx, chainID)
+			if err != nil {
+				return fmt.Errorf("could not get chain client for chain %d: %w", chainID, err)
+			}
+
+			ierc20, err := ierc20.NewIERC20(common.HexToAddress(tokenCFG.Address), chainClient)
```

Comment:
There's one edge case here when token address is `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE`. This is used to represent a native gas token (Ether) and therefore this address doesn't correspond to an actual smart contract.

In this edge case we should use 18 as `actualDecimals`

Reply:
```go
			// Check if the token is the gas token. SHOULD BE 18.
			if tokenCFG.Address == chain.EthAddress.String() {
				if tokenCFG.Decimals != 18 {
					return fmt.Errorf("decimals mismatch for token %s on chain %d: expected 18, got %d", tokenName, chainID, tokenCFG.Decimals)
				}
				continue
			}
```